### PR TITLE
Monkey trip mechanic removed from AI controller and added to brain. Adds organ based trait system.

### PR DESCRIPTION
**Pull request by itseasytosee**
## About The Pull Request
Alright, big exciting PR.

First big thing! Traits can now be tried to organs which are granted and removed from the mob with the organ.
The traits PRIMATIVE, LITERATE, CAN_STRIP, and ADVANCED_TOOL_USER have been removed from species inherent traits list, because they apply to the species brain in specific, not their whole body.
The baseline brain gives the traits LITERATE, CAN_STRIP, and ADVANCED_TOOL_USER, all species that had these traits have a baseline brain. 
Ashwalkers have been given a "primitive brain" which gives ADVANCEDTOOLUSER, CAN_STRIP, PRIMITIVE, and notably not literacy. These give them the same traits as before, but their privative nature and lack of reading followers their brain no matter which body they inhabit.

Monkeys have been given the "primate brain" which gives CAN_STRIP and PRIMITIVE. The monkey's brain now holds the logic for the monkey tripping mechanic instead of the AI datum which is a huge win. Monkeys with clients can also now toggle if they would like to be tripped by people, in order to better blend in. AI monkeys will always have tripping on. I also have implemented my qol monkey tripping changes from #69478 (the videos on that PR are still accurate)

What this means, is  a human's brain in a monkey's body will still be able to use tools, read, and even operate mechs. but will still have the other non mental monkey traits like being able to crawl though vents while naked. So lovely.

Conversely a monkey's brain (even a sentient one) that is put in a human body will still be illiterate, and will be unable to use complex tools.

Transforming into a monkey via genetics or other means will also change your brain into a monkey brain (this mechanic existed before now, it just wasn't applicable)
 
Some of this code might not be up to standard, please give me your corrections.

## Why It's Good For The Game
This is a pretty good step forward with species code, and the organ based trait system opens up many more possibilities for future PRs.
AI shouldn't introduce behavior the base mob (or in this case the mob's brain) isn't able to do. Armhulenn helped me out with how this could be implemented in a cool way. The trip qol stuff shouldn't change how you interact with monkeys, it just stops them from clambering over eachother and falling over at the slightest breeze. Sentient monkeys being able to choose whether or not they want to trip is based on the idea that players controlling mobs should be able to do anything that the mob could do, even if its mostly detrimental, But also allows them to not be total chumps in any combat scenario if they want.
so dope

## Changelog

:cl: itseasytosee
qol: Monkeys wont trip over each other anymore, and will only trip if you run into them with combat mode on.
qol: Player monkeys can now opt in to tripping like a regular monkey.
refactor: Stores monkey tripping in the brain, rather than the AI controller
refactor: Traits such as literacy and advanced tool use are now tied to the brain rather than the species 
/:cl:

### CODE REVIEW

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Review by MrMelbert — CHANGES_REQUESTED** [review]

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/species_types/lizardpeople.dm [hunk 1]
```diff
@@ -131,9 +128,8 @@ Lizard subspecies: ASHWALKERS
 	id = SPECIES_LIZARD_ASH
 	species_traits = list(MUTCOLORS,EYECOLOR,LIPS,HAS_FLESH,HAS_BONE)
 	mutantlungs = /obj/item/organ/internal/lungs/ashwalker
+	mutantbrain = /obj/item/organ/internal/brain/primitive
```
[the remark this answers is not in the document]
Comment (itseasytosee) [reply code/modules/mob/living/carbon/human/species_types/lizardpeople.dm:131]: Nope. Ashwalkers have a primitive brain, monkeys have a primate brain. Askwalkers are advanced tool users after all.

[the remark this answers is not in the document]
Comment (itseasytosee) [reply code/modules/mob/living/carbon/human/species_types/lizardpeople.dm:131]: I considered calling it an "askwalker" or "corrupted" brain if you think those names are better. 
The primitive brain just doesn't give you literacy and give you the "primitive" trait instead. (something ash walkers already had)

**Comment by itseasytosee** [comment]
That should be everything

**Comment by itseasytosee** [comment]
>Wait ashwalkers will be given the "tripping" action too yes? Seems weird

Nope. Ashwalkers have a primitive brain, monkeys have a primate brain. Askwalkers are advanced tool users after all.
I considered calling it an "ashwalker" or "corrupted" brain if you think those names are better.
The primitive brain just doesn't give you literacy and give you the "primitive" trait instead. (something ash walkers already had)

**Comment by itseasytosee** [comment]
oh fuck forgot to unregister the signal

**Comment by itseasytosee** [comment]
Ok, good to go.

**Comment by timothymtorres** [comment]
Something to consider:

https://github.com/tgstation/tgstation/blob/be0e9ee0f4ad853696157e7478047cc87fc0abbf/code/datums/mutations/passive.dm#L15-L33

I'm very glad to see this change.  I have been working on organs for a special DLC in the future and having the ability to assign traits to will be great.  I can already think of a number of uses.  (eyes having `NOFLASH`, stomach having `TRAIT_TOXINLOVER`, etc.)

**Comment by itseasytosee** [comment]
> Something to consider:
> 
> https://github.com/tgstation/tgstation/blob/be0e9ee0f4ad853696157e7478047cc87fc0abbf/code/datums/mutations/passive.dm#L15-L33
> 

This is kind of a difficult problem. It would be possible to add these traits to the targets brain (along side a necessary proc to update traits on brains, currently it only works on insertion.)
But instability cost is tied to the mob, and it would be really weird having the brain carry instability with it.
I really can't think of a good solution, but the issue has inspired me to add a update_traits() proc and a add_organ_trait() proc so situations where we want to add traits to organs on the fly can be done easily.

**Comment by itseasytosee** [comment]
Ok, now I'm happy with the state of the PR.

**Comment by tralezab** [comment]
So am I. Great work!

**Pull request merged**